backend/database.js

```diff
@@ -84,6 +84,46 @@ export default (sbp('sbp/selectors/register', {
       }
     })
   },
+  'backend/db/streamEntriesBetween': async function (startHash: string, endHash: string, offset: number): Promise<*> {
+    let prefix = '['
+    let isMet = false
+    let currentHEAD = endHash
+    let entry = await sbp('chelonia/db/getEntry', currentHEAD)
+    if (!entry) {
+      throw Boom.notFound(`entry ${currentHEAD} doesn't exist!`)
+    }
+    // NOTE: if this ever stops working you can also try Readable.from():
+    // https://nodejs.org/api/stream.html#stream_stream_readable_from_iterable_options
+    return new Readable({
+      async read (): any {
+        try {
+          entry = await sbp('chelonia/db/getEntry', currentHEAD)
+          const json = `"${strToB64(entry.serialize())}"`
+          this.push(prefix + json)
+          prefix = ','
+
+          if (currentHEAD === startHash) {
+            isMet = true
+          } else if (isMet) {
+            offset--
+          }
+
+          const previousHEAD = entry.message().previousHEAD
+          if (!currentHEAD || !previousHEAD || (isMet && !offset)) {
```

Comment:
I think this condition should be:

```js
          if (!previousHEAD || (isMet && offset <= 0)) {
```

Because:

1. How can `currentHEAD` ever be `null` here?
2. If an `offset` of `0` was passed in, then your last condition will fail, because `offset` will be `-1`, and `!-1` is `false`

Reply:
1. I think that's right. currentHEAD couldn't be null here.
2. Even though offset of 0 is passed in, it couldn't be -1 here because of the following condition.
```
if (currentHEAD === startHash) {
  isMet = true
} else if (isMet) {
  offset--
}
```

Reply:
Nice catch 👍